libs/packages/sam-formly/src/lib/formly/types/textarea.ts

```diff
@@ -68,6 +76,20 @@ export class FormlyFieldTextAreaComponent extends FieldType implements OnInit {
     } else {
       this.charactersRemaining = Math.max(0, this.to.maxLength - value.length);
     }
+    let regex = this.to.regexPattern;
+    if (!regex.test(value)) {
+      let words = value.split(' ');
+      for (let i in words) {
+        if (!regex.test(words[i])) {
+          console.log(words[i]);
+          words[i] = '<mark>' + words[i] + '</mark>';
+        }
```

Comment:
This strategy if fairly brittle and probably won't meet requirement for adding in advanced search. It looks like if the over all regex fails, this algorithm then goes to each space separated word and tests where it failed. If the whole input fails the test, it doesn't necessarily mean it was caused by an individual word, but rather a combination of those words. For example, `asphalt and pavement` makes sense but `asphalt and or and pavement` is a bit nonsensical, and fails not through individual word, but through word structure.